backend/apis/leads/list-leads.api.js

```diff
@@ -0,0 +1,113 @@
+const { app, databaseError, pool } = require("../../server");
+const mysql = require("mysql");
+const {
+  responseFullName,
+  responseBoolean,
+  responseDateWithoutTime
+} = require("../utils/transform-utils");
+
+app.get("/api/leads", (req, res, next) => {
+  const getLeads = mysql.format(`
+    SELECT
+      leads.id AS leadId,
+      leads.dateOfSignUp,
+      leads.firstContactAttempt,
+      leads.secondContactAttempt,
+      leads.thirdContactAttempt,
+      leads.inactivityReason,
+      leads.firstName,
+      leads.lastName,
+      leads.phone,
+      leads.smsConsent,
+      leads.zip,
+      leads.age,
+      leads.gender,
+      leads.clientId,
+      leads.isDeleted,
+      leads.dateAdded,
+      leads.dateModified,
+      leads.addedBy,
+      leads.modifiedBy,
+      created.firstName AS createdByFirstName,
+      created.lastName AS createdByLastName,
+      modified.firstName AS modifiedByFirstName,
+      modified.lastName AS modifiedByLastName,
+      JSON_ARRAYAGG(
+        JSON_OBJECT(
+          "serviceId", leadServices.serviceId,
+          "serviceName", services.serviceName
+        )
+      ) AS leadServices
+    FROM leads
+      INNER JOIN users created 
+        ON created.id = leads.addedBy
+      INNER JOIN users modified 
+        ON modified.id = leads.modifiedBy
+      INNER JOIN leadServices 
+        ON leadServices.leadId = leads.id
+      INNER JOIN services 
+        ON services.id = leadServices.serviceId
+    WHERE leads.isDeleted = false
+    GROUP BY leadServices.leadId;
```

Comment:
This SQL query should be paginated. We should also update the [docs for this api](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/master/api-docs/leads.md#get-all-leads) to be paginated.

I'm okay with the events api not being paginated because there likely won't be too many of those. But there will be thousands of leads and I think it would be irresponsible to not make the backend and UI paginated.

To make it paginated, see [here](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/master/backend/apis/clients/list-clients.api.js#L119) and [here](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/master/backend/apis/clients/list-clients.api.js#L124-L125).

Also, showing the total number of leads that match the search is another nice thing to add to the API. SELECT_FOUND_ROWS() is the way you can do that.